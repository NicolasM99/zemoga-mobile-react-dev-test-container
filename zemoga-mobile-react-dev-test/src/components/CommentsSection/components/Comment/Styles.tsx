import { StyleSheet } from 'react-native';

export const commentStyles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    backgroundColor: 'rgba(0,0,0,0.1)',
    borderRadius: 8,
    marginBottom: 8
  }
});
