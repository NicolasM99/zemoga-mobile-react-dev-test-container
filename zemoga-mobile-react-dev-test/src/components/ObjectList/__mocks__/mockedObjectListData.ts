export const mockedObjectList = {
  test: 'Test attribute',
  nested0: {
    nestedlvl0attribute: 'Nested level 0 attribute',
    nested1: {
      nestedlvl1attribute: 'Nested level 1 attribute'
    }
  }
};
