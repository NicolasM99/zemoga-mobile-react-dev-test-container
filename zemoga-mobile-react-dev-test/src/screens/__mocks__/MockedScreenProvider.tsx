import React from 'react';

import { NavigationContainer } from '@react-navigation/native';

import { PostsListProvider } from 'src/components/PostsList/context/PostsListContext';
import { InternetStatusProvider } from 'src/context/InternetStatusContext';

interface MockedScreenProviderProps {
  children: React.ReactNode;
}

const MockedScreenProvider: React.FC<MockedScreenProviderProps> = ({
  children
}: MockedScreenProviderProps) => (
  <NavigationContainer>
    <InternetStatusProvider>
      <PostsListProvider>{children}</PostsListProvider>
    </InternetStatusProvider>
  </NavigationContainer>
);

export default MockedScreenProvider;
