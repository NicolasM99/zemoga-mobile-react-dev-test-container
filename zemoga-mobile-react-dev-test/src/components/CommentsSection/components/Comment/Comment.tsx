import React, { FC } from 'react';

import { View } from 'react-native';

import Paragraph from 'src/components/Paragraph/Paragraph';
import Small from 'src/components/Small/Small';

import { commentStyles } from './Styles';
import { IComment } from '../../@types/commentsSection';

const Comment: FC<IComment> = ({ name = '', email = '', body = '' }: IComment) => {
  return (
    <View style={commentStyles.container}>
      <Small>{name}</Small>
      <Small>{email}</Small>
      <Paragraph style={{ marginTop: 8 }}>{body}</Paragraph>
    </View>
  );
};

export default Comment;
