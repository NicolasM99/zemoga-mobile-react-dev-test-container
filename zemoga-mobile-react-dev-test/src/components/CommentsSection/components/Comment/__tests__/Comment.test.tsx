import { render, screen } from '@testing-library/react-native';

import { mockedComment } from '../__mocks__/mockedCommentData';
import Comment from '../Comment';

describe('<Comment /> unit tests', () => {
  it('Should mount correctly the component', () => {
    render(<Comment {...mockedComment} />);
    expect(screen).toMatchSnapshot();
  });

  it('Should mount correctly the component with correct data', async () => {
    render(<Comment {...mockedComment} />);
    const commentName = await screen.findByTestId('commentName');
    const commentEmail = await screen.findByTestId('commentEmail');
    const commentBody = await screen.findByTestId('commentBody');
    expect(commentName.props.children).toEqual(mockedComment.name);
    expect(commentEmail.props.children).toEqual(mockedComment.email);
    expect(commentBody.props.children).toEqual(mockedComment.body);
  });
});
