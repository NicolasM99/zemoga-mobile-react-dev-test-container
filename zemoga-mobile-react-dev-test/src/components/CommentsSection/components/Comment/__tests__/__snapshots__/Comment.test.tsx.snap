// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> unit tests Should mount correctly the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.1)",
      "borderRadius": 8,
      "flex": 1,
      "marginBottom": 8,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 12,
      }
    }
    testID="commentName"
  >
    id labore ex et quam laborum
  </Text>
  <Text
    style={
      Object {
        "fontSize": 12,
      }
    }
    testID="commentEmail"
  >
    Eliseo@gardner.biz
  </Text>
  <Text
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 8,
      }
    }
    testID="commentBody"
  >
    laudantium enim quasi est quidem magnam voluptate ipsam eos
tempora quo necessitatibus
dolor quam autem quasi
reiciendis et nam sapiente accusantium
  </Text>
</View>
`;
